packages/gitbook/src/middleware.ts

```diff
@@ -332,27 +347,36 @@ function getInputURL(request: NextRequest): { url: URL; mode: URLLookupMode } {
         mode = 'multi-id';
     }
 
+    // When passing a x-gitbook-site-url header, this URL is used instead of the request URL
+    // to determine the site to serve.
+    const xGitbookSite = request.headers.get('x-gitbook-site-url');
+    if (xGitbookSite) {
+        mode = 'proxy';
+    }
```

Comment:
I think we should be setting the `url` using `xGitbookSite` here instead of doing it in `lookupSiteInProxy` because that lookup function should only be concerned with getting a `mode` and `url` and determine how to best serve the site.

Reply:
This is what I originally thought, but actually it'll cause trouble as we also use this URL as a representation of where the user currently is, which is not the case.